PayPalCheckout/src/main/java/com/paypal/android/checkout/PayPalClient.kt

```diff
@@ -1,44 +1,96 @@
 package com.paypal.android.checkout
 
-import android.app.Application
-import android.os.Build
-import androidx.annotation.RequiresApi
-import com.paypal.android.checkout.pojo.ErrorInfo
+import androidx.fragment.app.FragmentActivity
+import com.braintreepayments.api.BrowserSwitchClient
+import com.braintreepayments.api.BrowserSwitchResult
+import com.braintreepayments.api.BrowserSwitchStatus
+import com.paypal.android.checkout.error.PayPalError
 import com.paypal.android.core.CoreConfig
-import com.paypal.checkout.PayPalCheckout
-import com.paypal.checkout.approve.OnApprove
-import com.paypal.checkout.cancel.OnCancel
-import com.paypal.checkout.config.CheckoutConfig
-import com.paypal.checkout.createorder.CreateOrder
-import com.paypal.checkout.error.OnError
 
-@RequiresApi(Build.VERSION_CODES.M)
-class PayPalClient(application: Application, coreConfig: CoreConfig, returnUrl: String) {
+
+/**
+ * Use this client to approve an order with a [PayPalRequest].
+ */
+class PayPalClient internal constructor(
+    private val activity: FragmentActivity,
+    private val coreConfig: CoreConfig,
+    private val browserSwitchClient: BrowserSwitchClient,
+    private val browserSwitchHelper: BrowserSwitchHelper
+) {
+
+    /**
+     * Create a new instance of [PayPalClient].
+     *
+     * @param activity a [FragmentActivity]
+     * @param configuration a [CoreConfig] object
+     * @param urlScheme the custom URl scheme used to return to your app from a browser switch flow
+    */
+    constructor(
+        activity: FragmentActivity,
+        configuration: CoreConfig,
+        urlScheme: String
+    ) : this(activity, configuration, BrowserSwitchClient(), BrowserSwitchHelper(urlScheme))
+
+    private var browserSwitchResult: BrowserSwitchResult? = null
+    var listener: PayPalCheckoutListener? = null
+        set(value) {
+            field = value
+            browserSwitchResult?.also {
+                handleBrowserSwitchResult()
```

Comment:
This handles a browser switch immediately when the listener is set - this allows merchants to set a listener after their Activity has been resumed, and the browser switch has been delivered, and still receive the result. How should should this behavior be documented?